src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] supplyAndBuy = readData(fromFileName);
+        int result = processStatistics(supplyAndBuy[0], supplyAndBuy[1]);
+        writeData(toFileName, supplyAndBuy[0], supplyAndBuy[1], result);
+    }
+
+    private int[] readData(String fromFileName) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] data = line.split(",");
+                String operation = data[0];
+                int amount = Integer.parseInt(data[1]);
+
+                if (operation.equals("supply")) {
+                    totalSupply += amount;
+                } else if (operation.equals("buy")) {
+                    totalBuy += amount;
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading from file: " + fromFileName, e);
+        }
+
+        return new int[]{totalSupply, totalBuy};
+    }
+
+    private int processStatistics(int totalSupply, int totalBuy) {
+        return totalSupply - totalBuy;
+    }
+
+    private void writeData(String toFileName, int totalSupply, int totalBuy, int result) {
```

Comment:
The `writeData` method is well encapsulated and follows the single responsibility principle by only handling the writing of data to a file.